integrations/google_ai/src/haystack_integrations/components/generators/google_ai/gemini.py

```diff
@@ -100,12 +100,12 @@ def __init__(
         """
         genai.configure(api_key=api_key.resolve_value())
 
-        self._api_key = api_key
-        self._model_name = model
-        self._generation_config = generation_config
-        self._safety_settings = safety_settings
-        self._model = GenerativeModel(self._model_name)
-        self._streaming_callback = streaming_callback
+        self.api_key = api_key
+        self.model_name = model
+        self.generation_config = generation_config
+        self.safety_settings = safety_settings
+        self.model = GenerativeModel(self.model_name)
+        self.streaming_callback = streaming_callback
```

Comment:
I would make only the following public: `generation_config`, `safety_settings` and `streaming_callback`.
If I'm not mistaken, changing these attributes at runtime actually modifies the behavior of the component.
(this is not true for `model_name`, for example)

Reply:
I agree with that, but I also may want as a user to easily inspect the name of the model I'm using in a component when working with these in python and it would feel odd that I would need to access a private variable to do so, right?

Reply:
ok let's leave `model_name` public. What about `api_key`?

Reply:
Maybe in the spirit of minimizing changes let's leave it internal. But it is public in most of our other components. Perhaps a discussion on standardizing what should and shouldn't be public would be a good idea?